File: mirai-core/src/main/java/net/mamoe/mirai/network/protocol/tim/packet/PacketId.java
package net.mamoe.mirai.network.protocol.tim.packet;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * @author Him188moe
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
public @interface PacketId {
    /**
     * 获取用于识别的包 ID
     */
    String value();
}

File: mirai-core/src/jvmMain/kotlin/net/mamoe/mirai/network/protocol/tim/packet/PacketId.kt
package net.mamoe.mirai.network.protocol.tim.packet

/**
 * @author Him188moe
 */
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE)
annotation class PacketId(
        /**
         * 用于识别的包 ID
         */
        val value: String
)
